### PR TITLE
#ba 79 image service

### PR DESCRIPTION
**Pull request by hetec**
This update enables a user to add new or alter an existing image for a recipe. Maybe there could be a remaining bug, however I have tried the most possible combinations of actions performable by a user on the recipe component.

### CODE REVIEW

File: socobo-recipe-details-add.html [hunk 1]
```diff
@@ -168,7 +206,20 @@
...
+              <iron-icon icon="create"></iron-icon>
+            </div>
+            <iron-image
+              id="add_image"
+              class="recipeImg_image"
+              src="{{recipe.image}}"
+              sizing="cover"
+              preload="true"
+              fade
+              placeholder="http://localhost:8080/components/socobo-recipe/placeholder.png">
```
Comment (flashback2k14) [inline socobo-recipe-details-add.html]: this image path only works in development! please replace with ../placeholder.png

File: ImageService.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+function ImageService(imageCooserId){
+
+  var imageChooser = document.querySelector(imageCooserId);
+  var data = "";
+
+  this.encode = function (imageId) {
+    var file = imageChooser.files[0];
+    if(file === undefined || file === null){
+      data = "";
+      build(imageId,"http://localhost:8080/components/socobo-recipe/placeholder.png");
```
Comment (flashback2k14) [inline ImageService.js]: this image path only works in development! please replace with ../placeholder.png

File: socobo-recipe-details-edit.html [hunk 3]
```diff
@@ -167,8 +209,35 @@
...
+            </div>
+            <template is="dom-if" if="{{recipe.image}}">
+              <iron-image
+                id="edit_image"
+                class="recipeImg_image"
+                src="{{recipe.image}}"
+                sizing="cover"
+                preload="true"
+                fade
+                placeholder="http://localhost:8080/components/socobo-recipe/placeholder.png">
```
Comment (flashback2k14) [inline socobo-recipe-details-edit.html]: this image path only works in development! please replace with ../placeholder.png

File: socobo-recipe-details-edit.html [hunk 4]
```diff
@@ -167,8 +209,35 @@
...
+            </template>
+            <template is="dom-if" if="{{!recipe.image}}">
+              <iron-image
+                id="edit_placeholder"
+                class="recipeImg"
+                src="placeholder.png"
+                sizing="cover"
+                preload="true"
+                fade
+                placeholder="http://localhost:8080/components/socobo-recipe/placeholder.png">
```
Comment (flashback2k14) [inline socobo-recipe-details-edit.html]: this image path only works in development! please replace with ../placeholder.png

File: socobo-recipe-details-show.html [hunk 5]
```diff
@@ -129,6 +136,28 @@ <h1 id="title_default" class="default">{{recipe.desc}}</h1>
...
           <paper-material elevation="1" class="recipeImg">
+            <template is="dom-if" if="{{recipe.image}}">
+              <iron-image
+                      id="show_image"
+                      class="recipeImg_image"
+                      src="{{recipe.image}}"
+                      sizing="cover"
+                      preload="true"
+                      fade
+                      placeholder="http://localhost:8080/components/socobo-recipe/placeholder.png">
```
Comment (flashback2k14) [inline socobo-recipe-details-show.html]: this image path only works in development! please replace with ../placeholder.png

File: socobo-recipe-details-show.html [hunk 6]
```diff
@@ -129,6 +136,28 @@ <h1 id="title_default" class="default">{{recipe.desc}}</h1>
...
+            </template>
+            <template is="dom-if" if="{{!recipe.image}}">
+              <iron-image
+                      id="recipe_image"
+                      class="recipeImg"
+                      src="placeholder.png"
+                      sizing="cover"
+                      preload="true"
+                      fade
+                      placeholder="http://localhost:8080/components/socobo-recipe/placeholder.png">
```
Comment (flashback2k14) [inline socobo-recipe-details-show.html]: this image path only works in development! please replace with ../placeholder.png

File: socobo-recipe-overview.html [hunk 7]
```diff
@@ -131,12 +131,26 @@
       }
     },
     clicked : function (e){
-      this.$.showDialog.recipe = e.detail;
+      var image = new RecipeDao(this, this.user).findImage(e.detail);
+      var recipe = e.detail;
+      if(image !== undefined && image !== null && image !== ""){
+        recipe.image = image;
+      }else{
+        recipe.image = "http://localhost:8080/components/socobo-recipe/placeholder.png";
```
Comment (flashback2k14) [inline socobo-recipe-overview.html]: this image path only works in development! please replace with ../placeholder.png

File: socobo-recipe-overview.html [hunk 8]
```diff
@@ -131,12 +131,26 @@
...
       this.$.showDialog.toggleDialog();
     },
     edit : function (e){
-      this.$.editDialog.recipe = e.detail;
+      var image = new RecipeDao(this, this.user).findImage(e.detail);
+      var recipe = e.detail;
+      if(image !== undefined && image !== null && image !== ""){
+        recipe.image = image;
+      }else{
+        recipe.image = "http://localhost:8080/components/socobo-recipe/placeholder.png";
```
Comment (flashback2k14) [inline socobo-recipe-overview.html]: this image path only works in development! please replace with ../placeholder.png

**Comment by flashback2k14** [comment]
i.O. after change placerholder image path

**Comment by hetec** [comment]
Thanks! Will do

**Pull request merged**